app/services/ui_notifications/report_finished.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module UINotifications
+  class ReportFinished < UINotifications::Base
+    include Rails.application.routes.url_helpers
+
+    def initialize(composer, job_id)
+      @composer = composer
+      @job_id = job_id
+      @template = composer.template
+      super(User.current)
```

Comment:
This may be my preference, but I always prefer to call `super` before I apply any custom logic. Perhaps because in Java you have to and that's where I learned it, but also because `super()` may always set `@template` to some default. If there's a good reason `super()` shouldn't be called first, I'd expect a comment.

Reply:
I understand your concern but the code of the parent allows that because instance variables (`@composer`,`@job_id`, `@template`) are not defined. On the other hand, I see that could be a bad habit for every object language so I moved call at the begging.